src/guide/composition-api-provide-inject.md

```diff
@@ -1,120 +1,261 @@
 # Provide / Inject
 
-> This guide assumes that you have already read the [Composition API Introduction](composition-api-introduction.html) and [Reactivity Fundamentals](reactivity-fundamentals.html). Read that first if you are new to Composition API.
+> This guide assumes that you have already read [Provide / Inject](component-provide-inject.html), [Composition API Introduction](composition-api-introduction.html), and [Reactivity Fundamentals](reactivity-fundamentals.html).
 
 We can use [provide / inject](component-provide-inject.html) with the Composition API as well. Both can only be called during [`setup()`](composition-api-setup.html) with a current active instance.
 
-For example, if we want to provide a book name on the root component and inject it on the child component:
+## Scenario Background
 
-```js
-import { provide, inject } from 'vue'
+Let's assume that we want to rewrite the following code, which contains a `MyMap` component that provides a `MyMarker` component with the user's location, using the Composition API.
+
+```vue
+<!-- src/components/MyMap.vue -->
+<template>
+  <MyMarker />
+</template>
+
+<script>
+import MyMarker from './MyMarker.vue'
+
+export default {
+  components: {
+    MyMarker
+  },
+  provide: {
+    location: 'North Pole',
+    geolocation: {
+      longitude: 90,
+      latitude: 135
+    }
+  }
+}
+</script>
+```
+
+```vue
+<!-- src/components/MyMarker.vue -->
+<script>
+export default {
+  inject: ['location', 'longitude', 'latitude']
+}
+</script>
+```
+
+## Using Provide
+
+When using `provide` in `setup()`, we start by explicitly importing the method from `vue`. This allows us to define each property with its own invocation of `provide`.
+
+The `provide` function allows you to define the property through two parameters:
+
+1. The property's name (`<String>` type)
+2. The property's value
 
-const RootComponent = {
+Using our `MyMap` component, our provided values can be refactored as the following:
+
+```vue{7,14-20}
```

Comment:
Nice! I didn't know we could do this. But is it necessary here though, if `vue` is meant for the whole SFC syntax? It might also create more maintenance burden (one would need to update these numbers whenever the code is changed).

Reply:
I'm a little confused. Are you saying we shouldn't use line highlighting? Or are you recommending splitting it up into HTML and JS for easier maintenance?

Reply:
Oops sorry, I misread the code. Ignore me, all good :)